src/sentry/search/events/datasets/metrics_layer.py

```diff
@@ -0,0 +1,608 @@
+from __future__ import annotations
+
+from typing import Callable, Mapping, Optional, Union
+
+import sentry_sdk
+from snuba_sdk import AliasedExpression, Column, Condition, Function, Op
+
+from sentry.api.event_search import SearchFilter
+from sentry.exceptions import IncompatibleMetricsQuery, InvalidSearchQuery
+from sentry.search.events import constants, fields
+from sentry.search.events.builder import MetricsQueryBuilder
+from sentry.search.events.datasets import field_aliases, filter_aliases
+from sentry.search.events.datasets.metrics import MetricsDatasetConfig
+from sentry.search.events.types import SelectType, WhereType
+from sentry.snuba.metrics.naming_layer.mri import TransactionMRI
+from sentry.utils.numbers import format_grouped_length
+
+
+class MetricsLayerDatasetConfig(MetricsDatasetConfig):
+    def __init__(self, builder: MetricsQueryBuilder):
+        self.builder = builder
+
+    @property
+    def search_filter_converter(
+        self,
+    ) -> Mapping[str, Callable[[SearchFilter], Optional[WhereType]]]:
+        return {
+            constants.PROJECT_ALIAS: self._project_slug_filter_converter,
+            constants.PROJECT_NAME_ALIAS: self._project_slug_filter_converter,
+            constants.EVENT_TYPE_ALIAS: self._event_type_converter,
+            constants.TEAM_KEY_TRANSACTION_ALIAS: self._key_transaction_filter_converter,
+            "transaction.duration": self._duration_filter_converter,
+            "transaction": self._transaction_filter_converter,
+            "tags[transaction]": self._transaction_filter_converter,
+            constants.TITLE_ALIAS: self._transaction_filter_converter,
+        }
+
+    @property
+    def field_alias_converter(self) -> Mapping[str, Callable[[str], SelectType]]:
+        return {
+            constants.PROJECT_ALIAS: self._resolve_project_slug_alias,
+            constants.PROJECT_NAME_ALIAS: self._resolve_project_slug_alias,
+            constants.TEAM_KEY_TRANSACTION_ALIAS: self._resolve_team_key_transaction_alias,
+            constants.TITLE_ALIAS: self._resolve_title_alias,
+            "transaction": self._resolve_transaction_alias,
+            "tags[transaction]": self._resolve_transaction_alias,
+        }
+
+    def resolve_metric(self, value: str) -> str:
+        """Resolve to the MRI"""
+        metric_mri = constants.METRICS_MAP.get(value)
+        if metric_mri is None:
+            # Maybe this is a custom measurment?
+            for measurement in self.builder.custom_measurement_map:
+                if measurement["name"] == value and measurement["metric_id"] is not None:
+                    return measurement["mri_string"]
+        if metric_mri is None:
+            metric_mri = value
```

Comment:
```suggestion
            metric_mri = value
```
Not a big deal but I think this should still work. I thought it was kind of odd to have two of the same conditionals (another on line 52)